code/game/objects/items/storage/boxes.dm

```diff
@@ -609,6 +609,39 @@
 	playsound(loc, "rustle", 50, 1, -5)
 	user.visible_message("<span class='notice'>[user] hugs \the [src].</span>","<span class='notice'>You hug \the [src].</span>")
 
+/////clown box & honkbot assembly
+obj/item/storage/box/clown
+	name = "clown box"
+	desc = "A colorful cardboard box for the clown"
+	icon_state = "clownbox"
+	illustration = null
+	//foldable = null
+
+/obj/item/stack/sheet/cardboard/attackby(obj/item/stamp/clown/I, mob/user, params)
+	if(!istype(I, /obj/item/stamp/clown))
+		return ..()
+	to_chat(user, "<span class='notice'>You stamp the cardboard! Its a clown box! Honk!</span>")
+	var/obj/item/storage/box/clown/A = new
+	user.put_in_hands(A)
+	qdel(src)

```

Comment:
It uses up an entire stack of cardboard no matter how much is in the stack? That's not good.

Also, you should always drop something before deleting it to avoid GC issues.

Reply:
Destroy() takes care of removing it from inventories properly

Reply:
I'm aware of that, but for whatever reason, it's not enough, and you explicitly have to drop the item before deleting it, from my experience.